cmd/krel/cmd/release_notes.go

```diff
@@ -113,12 +170,124 @@ func runReleaseNotes() (err error) {
 	logrus.Infof("Using start tag %v", start)
 	logrus.Infof("Using end tag %v", tag)
 
-	_, err = releaseNotesFrom(start)
+	if releaseNotesOpts.createDraftPR {
+		if err = validateDraftPROptions(); err != nil {
+			return errors.Wrap(err, "validating PR command line options")
+		}
+	}
+
+	result, err := releaseNotesFrom(start)
 	if err != nil {
 		return errors.Wrapf(err, "generating release notes")
 	}
 
-	//TODO: implement PR creation for k-sigs/release-notes and k/sig-release
+	// Create RN draft PR
+	if releaseNotesOpts.createDraftPR {
+		if err := createDraftPR(tag, result); err != nil {
+			return errors.Wrap(err, "failed to create release notes draft PR")
+		}
+		return nil
+	}
+
+	switch releaseNotesOpts.Format {
+	case "json":
+		err = ioutil.WriteFile(filepath.Join(releaseNotesOpts.outputDir, "release-notes.json"), []byte(result.json), 0644)
+		if err != nil {
+			return errors.Wrap(err, "writing release notes JSON file")
+		}
+	case "markdown":
+		err = ioutil.WriteFile(filepath.Join(releaseNotesOpts.outputDir, "release-notes.md"), []byte(result.json), 0644)
+		if err != nil {
+			return errors.Wrap(err, "writing release notes markdown file")
+		}
+	default:
+		return errors.Errorf("%q is an unsupported format", releaseNotesOpts.Format)
+	}
+
+	// TODO: implement PR creation for k-sigs/release-notes
+	return nil
+}
+
+// validateDraftPROptions checks if we have all needed parameters to create the Release Notes PR
+func validateDraftPROptions() error {
+	if releaseNotesOpts.createDraftPR {
+		// Check if --draft-org is set
+		if releaseNotesOpts.draftOrg == "" {
+			logrus.Warn("cannot generate the Release Notes draft PR without --draft-org")
+		}
+
+		// Check if --draft-repo is set
+		if releaseNotesOpts.draftRepo == "" {
+			logrus.Warn("cannot generate the Release Notes draft PR without --draft-repo")
+		}
+
+		if releaseNotesOpts.draftOrg == "" || releaseNotesOpts.draftRepo == "" {
+			return errors.New("To generate the release notes draft you must define both --draft-org and --draft-repo")
+		}
+	}
+	return nil
+}
+
+// createDraftPR pushes the release notes draft to the users fork
+func createDraftPR(tag string, result *releaseNotesResult) error {
+	s, err := util.TagStringToSemver(tag)
+	if err != nil {
+		return errors.Wrapf(err, "no valid tag: %v", tag)
+	}
+
+	branchname := "release-notes-draft-" + tag
+
+	// checkout kubernetes/sig-release
+	sigReleaseRepo, err := git.CloneOrOpenGitHubRepo(releaseNotesOpts.sigreleaseForkPath, git.DefaultGithubOrg, git.DefaultGithubReleaseRepo, true)
+	if err != nil {
+		return errors.Wrap(err, "cloning k/sig-release")
+	}
+
+	// add the user's fork as a remote
+	err = sigReleaseRepo.AddRemote("userfork", releaseNotesOpts.draftOrg, releaseNotesOpts.draftRepo)
```

Comment:
As a follow-up (I can take care of that as well): If the fork already exists, then we could verify that it matches some pattern, like: `[git@]github.com:$ORG/$REPO`. If this is the case, then we're probably good to proceed

Reply:
See https://github.com/kubernetes/release/pull/1122